### ISSUE TITLE
String interpolation variables lead to `The string literal " appears 3 times in the file`

### ISSUE THREAD

**Issue by magro** [scalastyle/scalastyle#100]
For a file with log statements using string interpolation with the same variables scalastyle reports an error `The string literal " appears 3 times in the file`. E.g.

``` scala
Logger.info(s"foo $msg")
Logger.info(s"bar $msg")
Logger.info(s"baz $msg")
```

causes this (the log statements are used in different methods, only the same variable name is repeated). When string concatenation is used instead of string interpolation this is no longer reported:

``` scala
Logger.info("foo " + msg)
Logger.info("bar " + msg)
Logger.info("bar " + msg)
```

Another fix is to configure the `MultipleStringLiteralsChecker` with this parameter:

``` xml
<parameter name="ignoreRegex"><![CDATA[^"$]]></parameter>
```

Still it would be great to have the default behaviour fixed.

**Comment by matthewfarwell** [comment]
As a note - this is because the string interpolation is being split into its own AstNode.

**Comment by sokrahta** [comment]
+1
> Still it would be great to have the default behaviour fixed.

**Comment by alaendle** [comment]
Wouldn't it be better to have a mixture between the suggested and the commited workaround - just adding the '?'. This seems to cover more false positives.
```xml
<parameter name="ignoreRegex"><![CDATA[^""?$]]></parameter>
```

**Comment by matthewfarwell** [comment]
This isn't simple to fix in the code, because by the time it gets to scalastyle the parser has already done the deconstruction of s"${foo}" into "" + foo + "".

**Comment by baatout** [comment]
+1

**Comment by lediur** [comment]
Has there been any progress on this?

**Comment by matthewfarwell** [comment]
@lediur Yes, there has. I've been playing around with scala meta as a replacement for scalariform, and I've just implemented the MultipleStringLiteralsChecker, and I can avoid this problem. If you want to see more, please look at this commit: https://github.com/matthewfarwell/scalastyle/commit/e5e628ce9087208269abdad723655c4c3d44b856

Please note that scalameta is an experiment, and I don't know when it will be ready, if ever. But it looks like a reasonable way to go.
